### PR TITLE
core(images): cleanup ImageUsage to match other *Elements artifacts

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Cleans up the `ImageUsage` artifact to remove unnecessary network record and align with other `*Elements` artifacts.

**Related Issues/PRs**
#6747

### CODE REVIEW

File: lighthouse-core/test/audits/image-aspect-ratio-test.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ function generateRecord(url = 'https://google.com/logo.png', mimeType = 'image/p
 
 function generateImage(clientSize, naturalSize, networkRecord, props, src = 'https://google.com/logo.png') {
   Object.assign(networkRecord || {}, {url: src});
-  const image = {src, networkRecord};
+  const image = {src, ...networkRecord};
```
Comment (brendankenny) [inline lighthouse-core/test/audits/image-aspect-ratio-test.js]: wait, is this just for `resourceSize` and `mimeType`? :P

File: types/artifacts.d.ts [hunk 2]
```diff
@@ -239,14 +239,8 @@ declare global {
...
         };
-        networkRecord?: {
-          url: string;
-          resourceSize: number;
-          startTime: number;
-          endTime: number;
-          responseReceivedTime: number;
-          mimeType: string;
-        };
+        resourceSize?: number;
```
Comment (brendankenny) [inline types/artifacts.d.ts]: this default's to 0 so no need to be optional?

File: lighthouse-core/gather/gatherers/image-elements.js [hunk 3]
```diff
@@ -171,13 +164,16 @@ class ImageUsage extends Gatherer {
 
     const imageUsage = [];
     for (let element of elements) {
-      // link up the image with its network record
-      element.networkRecord = indexedNetworkRecords[element.src];
+      // Pull some of our information directly off the network record.
+      const networkRecord = indexedNetworkRecords[element.src] || {};
+      element.mimeType = networkRecord.mimeType;
+      const {resourceSize = 0, transferSize = 0} = networkRecord;
+      element.resourceSize = Math.min(resourceSize, transferSize);
```
Comment (brendankenny) [inline lighthouse-core/gather/gatherers/image-elements.js]: we didn't have one before, but mind putting a comment on this about the goal/why it's ok to fall back to `transferSize`?

File: lighthouse-core/test/audits/image-aspect-ratio-test.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/test/audits/image-aspect-ratio-test.js]: removed

File: types/artifacts.d.ts (hunk 2 above)
Comment (patrickhulce) [reply types/artifacts.d.ts]: done

File: lighthouse-core/gather/gatherers/image-elements.js (hunk 3 above)
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/image-elements.js]: sure, done

File: types/artifacts.d.ts [hunk 4]
```diff
@@ -224,31 +224,33 @@ declare global {
...
 
-      export interface SingleImageUsage {
+      export interface ImageElement {
         src: string;
-        clientWidth: number;
-        clientHeight: number;
+        /** The displayed width of the image, uses img.width when available falling back to clientWidth. See https://codepen.io/patrickhulce/pen/PXvQbM for examples. */
+        displayedWidth: number;
+        /** The displayed height of the image, uses img.height when available falling back to clientHeight. See https://codepen.io/patrickhulce/pen/PXvQbM for examples. */
+        displayedHeight: number;
```
Comment (Hoten) [inline types/artifacts.d.ts]: are these set when the image is from css?

Comment (patrickhulce) [reply types/artifacts.d.ts]: yes it falls back to `clientHeight`/`clientWidth` in that case

**Pull request merged**